test/box-luatest/gh_7894_export_iproto_constants_and_features_test.lua

```diff
@@ -14,170 +14,170 @@ g.after_all(function(cg)
     cg.server:drop()
 end)
 
-local reference_table = {
-    -- `IPROTO_FLAGS` bitfield constants enumeration.
-    flag = {
-        COMMIT = 0x01,
-        WAIT_SYNC = 0x02,
-        WAIT_ACK = 0x04,
-    },
-
-    -- `iproto_key` enumeration.
-    key = {
-        REQUEST_TYPE = 0x00,
-        SYNC = 0x01,
-        REPLICA_ID = 0x02,
-        LSN = 0x03,
-        TIMESTAMP = 0x04,
-        SCHEMA_VERSION = 0x05,
-        SERVER_VERSION = 0x06,
-        GROUP_ID = 0x07,
-        TSN = 0x08,
-        FLAGS = 0x09,
-        STREAM_ID = 0x0a,
-        SPACE_ID = 0x10,
-        INDEX_ID = 0x11,
-        LIMIT = 0x12,
-        OFFSET = 0x13,
-        ITERATOR = 0x14,
-        INDEX_BASE = 0x15,
-        FETCH_POSITION = 0x1f,
-        KEY = 0x20,
-        TUPLE = 0x21,
-        FUNCTION_NAME = 0x22,
-        USER_NAME = 0x23,
-        INSTANCE_UUID = 0x24,
-        CLUSTER_UUID = 0x25,
-        VCLOCK = 0x26,
-        EXPR = 0x27,
-        OPS = 0x28,
-        BALLOT = 0x29,
-        TUPLE_META = 0x2a,
-        OPTIONS = 0x2b,
-        OLD_TUPLE = 0x2c,
-        NEW_TUPLE = 0x2d,
-        IPROTO_AFTER_POSITION = 0x2e,
-        IPROTO_AFTER_TUPLE = 0x2f,
-        DATA = 0x30,
-        ERROR_24 = 0x31,
-        METADATA = 0x32,
-        BIND_METADATA = 0x33,
-        BIND_COUNT = 0x34,
-        IPROTO_POSITION = 0x35,
-        SQL_TEXT = 0x40,
-        SQL_BIND = 0x41,
-        SQL_INFO = 0x42,
-        STMT_ID = 0x43,
-        REPLICA_ANON = 0x50,
-        ID_FILTER = 0x51,
-        ERROR = 0x52,
-        TERM = 0x53,
-        VERSION = 0x54,
-        FEATURES = 0x55,
-        TIMEOUT = 0x56,
-        EVENT_KEY = 0x57,
-        EVENT_DATA = 0x58,
-        TXN_ISOLATION = 0x59,
-        VCLOCK_SYNC = 0x5a,
-    },
```

Comment:
Why do you move this code?

Reply:
Please see this [discussion](https://github.com/tarantool/tarantool/pull/7994#discussion_r1040866594).

Reply:
> This way, you can use it across server restarts or in different tests, if needed.

There is only one test that can use this table — what's the point? I also don't understand what you mean by using this table across server restarts.

Reply:
One might add more tests or check the table after restart although I admit it's unlikely with this particular table. Still, I don't see any point in bloating the diff when you can easily pass the table in `exec` parameters without making the code any more complicated.

BTW I just realized that for some reason you check that `box.iproto.send` equals `box.iproto.send`. What's the point? Why not check the constants explicitly?

```Lua
t.assert_equals(box.iproto.type, reference_table.type)
t.assert_equals(box.iproto.key, reference_table.key)
...
```

Reply:
Good point. I suggest we iterate over `box.iproto` and check that corresponding non-{function, thread, userdata} type values exist in the reference table.